src/lib/crypt_ops/crypto_rand_numeric.c

```diff
@@ -16,21 +49,9 @@
 int
 crypto_rand_int(unsigned int max)
 {
-  unsigned int val;
-  unsigned int cutoff;
   tor_assert(max <= ((unsigned int)INT_MAX)+1);
-  tor_assert(max > 0); /* don't div by 0 */
 
-  /* We ignore any values that are >= 'cutoff,' to avoid biasing the
-   * distribution with clipping at the upper end of unsigned int's
-   * range.
-   */
-  cutoff = UINT_MAX - (UINT_MAX%max);
-  while (1) {
-    crypto_rand((char*)&val, sizeof(val));
-    if (val < cutoff)
-      return val % max;
-  }
+  return (int)crypto_rand_uint(max);
```

Comment:
Hmm, this does not look right. Why don't we use the macro here and instead we are casting from unsigned to signed?

Reply:
The macro only works on unsigned outputs; this function needs to return a signed output.  I'm adding a fixup as 2bf39d7209c5b7 to explain this better.

Reply:
Ah makes sense. LGTM.